windows/sidecar.ps1

```diff
@@ -0,0 +1,372 @@
+                            @{
+                                name = "EndpointPolicy"
+                                value = @{
+                                    Type = "OutBoundNAT"
+                                    ExceptionList = @(
+                                        $clusterCIDR
+                                        $clusterServiceCIDR
+                                        $networkMetadataObj.SubnetCIDR
+                                    )
+                                }
+                            }
+                            @{
+                                name = "EndpointPolicy"
+                                value = @{
+                                    Type = "ROUTE"
+                                    NeedEncap = $true
+                                    DestinationPrefix = $clusterServiceCIDR
+                                }
+                            }
+                            @{
+                                name = "EndpointPolicy"
+                                value = @{
+                                    Type = "ROUTE"
+                                    NeedEncap = $true
+                                    DestinationPrefix = $networkMetadataObj.AddressCIDR
+                                }
+                            }
+                        )
+                    }
+                    $cniInfo = @{
+                        Mode = "l2bridge"
+                        Interface = "cbr0"
+                    }
+                }
+            }
+
+            @{
+                name = $cniInfo.Interface
+                cniVersion = "0.2.0"
+                plugins = @(
+                    @{
+                        type = "flannel"
+                        capabilities = @{
+                            dns = $true
+                        }
+                        delegate = $cniConfDelegate
+                    }
+                )
+            } | ConvertTo-Json -Compress -Depth 32 | Out-File -Encoding ascii -Force -FilePath "c:\host\etc\cni\net.d\10-flannel.conflist"
+
+        }
+
+    }
+
+}
+
+# output cni informantion for kube-proxy
+$cniInfo | ConvertTo-Json -Compress -Depth 32 | Out-File -Encoding ascii -Force -FilePath "c:\host\run\cni-info.json"
+Log-Info "Outputed CNI info"
+
+# start cni management
+if ($networkConfigObj) 
+{
+    switch ($networkConfigObj.plugin) 
+    {
+        "flannel" {
+            $type = $null
+            $vni = $null
+            $port = $null
+            try {
+                $options = $networkConfigObj.options
+                $type = $options.flannel_backend_type
+                $vni = $options.flannel_backend_vni
+                $port = $options.flannel_backend_port
+
+                if (-not $type) {
+                    $type = "vxlan"
+                }
+                if (-not $vni) {
+                    $vni = 4096
+                } else {
+                    $vni = [int]$vni
```

Comment:
We should comment that windows doesn't support changing vni for now.

Reply:
Commented